### ISSUE TITLE
use quarkus.live-reload.password as password on devui in remote mode?

### ISSUE THREAD

**Issue by maxandersen** [quarkusio/quarkus#44464]
### Description

currently devui is disabled in remotedev mode because devmode potentially can leak config/secrets

is there a reason we shouldn't allow user to explicit active devmode UI but then enforce basic auth access to devui using the `quarkus.live-reload.password` property (or some other property) ?

**Comment by sberyozkin** [comment]
Indeed, perhaps DevUI endpoint can have the same authentication support that the management endpoint has, in which case it won't be restricted to Basic authentication only.

**Comment by cescoffier** [comment]
@sberyozkin The dev ui is not on the management interface, so we cannot use this approach. We would need to setup authentication for the dev UI on the main HTTP server.

**Comment by sberyozkin** [comment]
Sure, thanks @cescoffier. 
If it is on the main server then it is already possible to have a path based authentication, https://quarkus.io/guides/security-authentication-mechanisms#use-http-security-policy-to-enable-path-based-authentication.
So maybe, the endpoint supporting remote devmode can indeed get protected with basic authentication like any other endpoint on the main server can ? CC @maxandersen

**Comment by mkouba** [comment]
> [@sberyozkin](https://github.com/sberyozkin) The dev ui is not on the management interface, so we cannot use this approach. We would need to setup authentication for the dev UI on the main HTTP server.

@cescoffier Do you happen to know why DevUI is not part of management interface?

**Comment by phillip-kruger** [comment]
The management interface is opt-in. So by default the Dev UI is not on there. We can maybe looking at moving it to the management interface when the management interface is enabled. I am not sure how much work that would be, but should be possible. Then we can say that to use this in remote dev, you have to enable management interface. We will also need some changes to enable dev ui for remote dev.
